src/handle_swap_sign_transaction.c

```diff
@@ -71,6 +93,28 @@ bool copy_transaction_parameters(create_transaction_parameters_t* sign_transacti
     G_swap_sign_return_value_address = &sign_transaction_params->result;
     // Commit the values read from exchange to the clean global space
 
+    // if destination_address_extra_id is given, we use the first byte to determine if we use the
+    // normal swap protocol, or the one for cross-chain swaps
+    if (destination_address_extra_data[0] == 0) {
```

Comment:
Just minor, but personally, I prefer using `switch` rather than several `else if`... I find it simpler for readability.
But this is minor, and just my opinion :wink:

Reply:
Fine with me, done